### DISCUSSION TITLE
question about project drafts vs regular projects

### DISCUSSION THREAD

**Discussion by bbc5008** [sadieblue/agilityblue.workspace-scripts#35] [Q&A] [answered]
Hello All,
I am working on an issue where TaskEmailMessages are continuing to append to closed tasks.  This is because our clients tend to never create new email threads and just continue to reply to existing threads for new tasks.  My workflow on fixing this with a script is to create a trigger event for new TaskEmailMessages that checks if the parent task is closed or not.  If the parent task has been closed, i would like to create a new draft project identical to the one the system creates when a new email is received.  my question is how do I create a project so that the system recognizes it as a draft?

**Comment by bmward80** [comment, marked as answer]
Hey, @bbc5008 ,

The project data model has a `ProjectTypeId` property that defaults to 1 (normal project). All you need to do is change that to a 2 to make the project a draft. If you're interested in learning the different types, you can use the `Get-ProjectTypes` cmdlet:

```powershell
# List the Project Types available:
(Get-ProjectTypes).Collection | ForEach-Object {
    Write-Output "$($_.ProjectTypeId) = $($_.Name)"
}
```

Outputs:
```
1 = Normal
2 = Draft
3 = Template
```

The bare minimum for creating a draft project:

```powershell
$project = @{
  Description   = "A project created by a script"
  ProjectTypeId = 2
}

$myDraftProject = Add-Project $project

Write-Output "Created a $($myDraftProject.ProjectTypeName.ToLower()) project (Id = $($myDraftProject.ProjectId))"
```

Outputs something similar to:
```
Created a draft project (Id = 2024100000002)
```

I hope this helps,
Brandon

**Reply by bbc5008** [reply]
many thanks.  this is very helpful
